app/models/metasploit_data_models/module_run.rb

```diff
@@ -0,0 +1,194 @@
+  SUCCEED     = 'succeeded'
+  # Marks the run as having not run successfully
+  FAIL        = 'failed'
+  # Marks the module as having had a runtime error
+  ERROR       = 'error'
+  # {ModuleRun} objects will be validated against these statuses
+  VALID_STATUSES = [SUCCEED, FAIL, ERROR]
+
+
+  #
+  # Attributes
+  #
+
+  # @!attribute [rw] attempted_at
+  #   The date/time when this module was run
+  #   @return [Datetime]
+
+  # @!attribute [rw] fail_detail
+  #   Arbitrary information captured by the module to give in-depth reason for failure
+  #   @return [String]
+
+  # @!attribute [rw] fail_reason
+  #   One of the values of the constants in {Msf::Module::Failure}
+  #   @return [String]
+
+  # @!attribute [rw] module_name
+  #   The Msf::Module#fullname of the module being run
+  #   @return [String]
+
+  # @!attribute [rw] port
+  #   The port that the remote host was attacked on, if any
+  #   @return [Fixnum]
+
+  # @!attribute [rw] proto
+  #   The name of the protocol that the host was attacked on, if any
+  #   @return [String]
+
+  # @!attribute [rw] session_id
+  #   The {Mdm::Session} that this was run with, in the case of a post module. In exploit modules, this field will
+  #   remain null.
+  #   @return [Datetime]
+
+  # @!attribute [rw] status
+  #   The result of running the module
+  #   @return [String]
+
+  # @!attribute [rw] username
+  #   The name of the user running this module
+  #   @return [String]
+
+
+
+  #
+  # Associations
+  #
+
+
+
+  # @!attribute [rw] loots
+  #  The sweet, sweet loot taken by this module_run
+  #
+  #  @return [ActiveRecord::Relation<Mdm::Loot>]
+  has_many :loots,
+           class_name: 'Mdm::Loot',
+           inverse_of: :module_run
+
+
+  # @!attribute [rw] spawned_session
+  #
+  #  The session created by running this module.
+  #  Note that this is NOT the session that modules are run on.
+  #
+  #  @return [Mdm::Session]
+  has_one :spawned_session,
+             class_name: 'Mdm::Session',
+             inverse_of: :originating_module_run
+
+
+  # @!attribute [rw] target_session
+  #
+  #  The session this module was run on, if any.
+  #  Note that this is NOT a session created by this module run
+  #  of exploit modules.
+  #
+  #  @return [Mdm::Session]
+  belongs_to :target_session,
+             class_name: 'Mdm::Session',
+             foreign_key: :session_id,
+             inverse_of: :target_module_runs
+
+
+
+  # @!attribute [rw] trackable
+  #
+  #  A polymorphic association that is tracked as being related to this module run.
+  #  {Mdm::Host} and {Mdm::Vuln} can each have {ModuleRun} objects.
+  #
+  #  @return [Mdm::Host, Mdm::Vuln]
+  belongs_to :trackable, polymorphic: true
+
+
+  # @!attribute [rw] user
+  #
+  #  The user that launched this module
+  #
+  #  @return [Mdm::User]
+  belongs_to :user,
+             class_name:  'Mdm::User',
+             foreign_key: 'user_id',
+             inverse_of: :module_runs
+
+
+
+  #
+  # Validations
```

Comment:
If I use method validations with `validate` and attribute validations of `validates`, I normally separate the too DSLs into subsections as it makes it easier to stops if I added or left off an `s` in the wrong place as Rails won't complain normally:

```ruby
#
#
# Validations
#
#

#
# Method Validations
#

# Can't save without information on what module has run
validate :module_information_is_present

# spawned_session is only valid for *exploit modules*
validate :no_spawned_session_for_non_exploits_except_logins

# target_session is only valid for *non-exploit modules*
validate :no_target_session_for_exploits

#
# Attribute Validations
#

# When the module was run
validates :attempted_at,
            presence: true

# Result of running the module
validates :status,
           inclusion: VALID_STATUSES
```

The Validations section increases one level to show it is encompassing the Method Validations and Attribute Validations sections.  Method validations are ordered based on the method name while attribute validations are ordered based on the attribute name.  (So the first argument to the DSL in each case.)

Also, I don't want to discourage comments, but I don't think comments on `validate` or `validates` go into the YARD docs.   When using `validate`, the comments should go on the method itself and when using `validates` the comments should go on the attribute.  Conditions on the attribute value can be `@note` or be mentioned in the `@param` docs for the writer method.